### PR TITLE
[WIP] Scheduler improvements.

### PR DESCRIPTION
**Pull request by brndnmtthws**
- Added additional configuration parameters for controlling scheduler
   behaviour:
   - mesos.offer.filter.seconds
   - mesos.offer.expiry.multiplier
   - mesos.prefer.reserved.resources
 - Implemented combining of reserved & unreserved resources
 - Improved Docker support, added support for running Storm inside
   containers. This also introduces the mesos.container.docker.image
   config param
 - Added unit tests
 - Improved port handling, especially with regard to logviewer
 - Code cleanup/style fixes
 - Implemented filters & reviveOffers()

NOTE: this isn't quite ready to merge yet.

### CODE REVIEW

File: src/main/storm/mesos/LaunchTask.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.apache.mesos.Protos;
+
+import static storm.mesos.PrettyProtobuf.offerToString;
+import static storm.mesos.PrettyProtobuf.taskInfoToString;
+
+class LaunchTask {
+  public final Protos.TaskInfo task;
+  public final Protos.Offer offer;
+
```
Comment (sargun) [inline src/main/storm/mesos/LaunchTask.java]: Any reason why these don't have getters? I realize that they're final, but as a matter of "good convention" getters are good.

File: src/test/storm/mesos/MesosNimbusTest.java [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+            Arrays.asList(
+                buildScalarResource("cpus", cpus),
+                buildScalarResource("mem", mem),
+                buildRangeResource("ports", portBegin, portEnd)
+            )
+        )
+        .build();
+  }
+
+  private Offer buildOfferWithReservation(double cpus, double mem, double reservedCpu, double reservedMem) {
```
Comment (sargun) [inline src/test/storm/mesos/MesosNimbusTest.java]: Please add a test to offer resources with a dynamic reservation. With principals that you're allowed to use and that you're not allowed to use.

**Comment by erikdw** [comment]
@brndnmtthws : please do the move of the code dirs as a separate change to the refactoring within the existing code.  Otherwise it's *really* hard to tell what has actually been changed.

File: src/test/storm/mesos/MesosNimbusTest.java (hunk 2 above)
Comment (brndnmtthws) [inline src/test/storm/mesos/MesosNimbusTest.java]: Done.

**Comment by brndnmtthws** [comment]
Okay, the PR should be a little easier to read now.

File: src/main/storm/mesos/LaunchTask.java (hunk 1 above)
Comment (brndnmtthws) [inline src/main/storm/mesos/LaunchTask.java]: Fair enough. I'll add 'em.

**Comment by brndnmtthws** [comment]
Closing this in favour of #65.

**Pull request closed without merging**